android/java/org/chromium/chrome/browser/brave_news/CardBuilderFeedCard.java

```diff
@@ -0,0 +1,1412 @@
+
+            setTextFromFeed(title, TITLE, index);
+
+            final FeedItemMetadata itemData = getItemData(index);
+
+            view.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    // @TODO alex refactor this with listener in BraveNewTabPageLayout
+                    BraveActivity.getBraveActivity().setNewsFeedScrollPosition(mPosition);
+                    TabUtils.openUrlInSameTab(itemData.url.url);
+                }
+            });
+
+            title.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    // @TODO alex refactor this with listener in BraveNewTabPageLayout
+                    BraveActivity.getBraveActivity().setNewsFeedScrollPosition(mPosition);
+                    TabUtils.openUrlInSameTab(itemData.url.url);
+                }
+            });
+
+            title.setOnLongClickListener(new View.OnLongClickListener() {
+                @Override
+                public boolean onLongClick(View v) {
+                    showBottomSheetDialog(
+                            itemData.url.url, itemData.publisherId, itemData.publisherName);
+                    return true;
+                }
+            });
+
+        } catch (Exception e) {
+            Log.e(TAG, "Exception addElementsToSingleLayout: " + e.getMessage());
+        }
+    }
+
+    private void showBottomSheetDialog(String urlString, String publisherId, String publisherName) {
+        BraveNewsBottomSheetDialogFragment bottomSheetDialog =
+                BraveNewsBottomSheetDialogFragment.newInstance();
+        Bundle bundle = new Bundle();
+        bundle.putString("url", urlString);
+        bundle.putString("publisherId", publisherId);
+        bundle.putString("publisherName", publisherName);
+        bottomSheetDialog.setArguments(bundle);
+        bottomSheetDialog.setController(mBraveNewsController);
+
+        bottomSheetDialog.show(((ChromeActivity) mActivity).getSupportFragmentManager(),
+                "brave_news_bottom_sheet_dialog_fragment");
+    }
+
+    private void setTextFromFeed(TextView textView, int type, int index) {
+        try {
+            FeedItemMetadata itemData = getItemData(index);
+            if (itemData != null) {
+                setText(itemData, textView, type);
+                setListeners(textView, itemData.url.url, mCreativeInstanceId, mIsPromo);
+            }
+
+        } catch (Exception e) {
+            Log.e(TAG, " exceptionscaught e:" + e.getMessage());
+        }
+    }
+
+    private FeedItemMetadata getItemData(int index) {
+        FeedItemMetadata itemData = null;
+        try {
+            // FeedItem feedItem = mNewsItem.getFeedItem();
+            List<FeedItemCard> feedItemsCard = mNewsItem.getFeedItems();
+            DisplayAd displayAd = mNewsItem.getDisplayAd();
```

Comment:
Is this variable being used? I can't see that it is.

Reply:
yeah it's not anymore. removing